core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.catalyst.analysis.NoSuchDatabaseException
+import org.apache.spark.sql.catalyst.catalog.{CatalogTable, SessionCatalog, TiSessionCatalog}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.types.{MetadataBuilder, StringType, StructType}
+import org.apache.spark.sql.{AnalysisException, Row, SparkSession, TiContext}
+
+import scala.collection.mutable.ArrayBuffer
+
+case class TiShowTablesCommand(tiContext: TiContext, delegate: ShowTablesCommand)
+    extends TiCommand(delegate) {
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val db = delegate.databaseName.getOrElse(tiCatalog.getCurrentDatabase)
+    // Show the information of tables.
+    val tables =
+      delegate.tableIdentifierPattern
+        .map(tiCatalog.listTables(db, _))
+        .getOrElse(tiCatalog.listTables(db))
+    tables.map { tableIdent =>
+      val database = tableIdent.database.getOrElse("")
+      val tableName = tableIdent.table
+      val isTemp = tiCatalog.isTemporaryTable(tableIdent)
+      if (delegate.isExtended) {
+        val information = tiCatalog.getTempViewOrPermanentTableMetadata(tableIdent).simpleString
+        Row(database, tableName, isTemp, s"$information\n")
+      } else {
+        Row(database, tableName, isTemp)
+      }
+    }
+  }
+}
+
+case class TiDescribeTablesCommand(tiContext: TiContext, delegate: DescribeTableCommand)
+    extends TiCommand(delegate) {
+  override val output: Seq[Attribute] = Seq(
```

Comment:
Why not simply use `delegate`'s output?

Reply:
I added nullable column in describe table command :) I would like to add more attributes that is not allowed in Hive but promising in TiDB~

Reply:
I added nullable column in describe table command :) I would like to add more attributes that is not allowed in Hive but promising in TiDB~